packages/affine/shared/src/services/doc-mode-service.ts

```diff
@@ -4,18 +4,20 @@ import type { Container } from '@blocksuite/global/di';
 
 import { Extension, StdIdentifier } from '@blocksuite/block-std';
 import { createIdentifier } from '@blocksuite/global/di';
-import { type Disposable, Slot } from '@blocksuite/global/utils';
+import { type Disposable, noop, Slot } from '@blocksuite/global/utils';
 
 const DEFAULT_MODE: DocMode = 'page';
 
 export interface DocModeProvider {
-  setMode: (mode: DocMode, docId?: string) => void;
-  getMode: (docId?: string) => DocMode;
-  toggleMode: (docId?: string) => DocMode;
-  onModeChange: (
+  setPrimaryMode: (mode: DocMode, docId: string) => void;
```

Comment:
Please add some comments here to explain when to use primary mode and when to use editor mode.